source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -77,6 +92,14 @@ void FilterConfigPerRoute::merge(const FilterConfigPerRoute& src) {
 void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callbacks) {
   Http::PassThroughFilter::setDecoderFilterCallbacks(callbacks);
   decoding_state_.setDecoderFilterCallbacks(callbacks);
+  const Envoy::StreamInfo::FilterStateSharedPtr& filter_state =
+      callbacks.streamInfo().filterState();
+  if (!filter_state->hasData<ExtProcStreamStats>(ExtProcStreamStatsName)) {
+    filter_state->setData(ExtProcStreamStatsName, std::make_shared<ExtProcStreamStats>(),
```

Comment:
I might be missing something here but why are we storing this in stream state vs just as a member of on the Filter? The filter itself is scoped to a stream

Reply:
In out use case the stream stats is for logging, we want to share the state with other filter for visibility. If it is a member of filter class, we won't be able to access the it from other filter.

Reply:
How about ExtProcLoggingInfo? CacheFilter defines a similarly-purposed FilterState object for logging/per-request metrics called [CacheFilterLoggingInfo](https://github.com/envoyproxy/envoy/blob/0d1781492579cfa0791babc010bef2555f30d31c/source/extensions/filters/http/cache/cache_filter_logging_info.h#L95).

ExtProcCallResult matches the current payload but I wonder if it might not age well if we start adding other information for logging as well, e.g. some form of status flags, etc.